atlasdb-api/src/main/java/com/palantir/atlasdb/transaction/api/TransactionSerializableConflictException.java

```diff
@@ -16,25 +16,43 @@
 package com.palantir.atlasdb.transaction.api;
 
 import com.palantir.atlasdb.keyvalue.api.TableReference;
+import com.palantir.logsafe.Arg;
+import com.palantir.logsafe.Safe;
+import com.palantir.logsafe.SafeLoggable;
+import java.util.List;
 
-public class TransactionSerializableConflictException extends TransactionFailedRetriableException {
+public class TransactionSerializableConflictException extends TransactionFailedRetriableException
+        implements SafeLoggable {
     private static final long serialVersionUID = 1L;
 
     private final TableReference conflictingTable;
+    private final List<Arg<?>> args;
 
-    public TransactionSerializableConflictException(String message, TableReference conflictingTable) {
+    public TransactionSerializableConflictException(
+            String message, TableReference conflictingTable, List<Arg<?>> args) {
         super(message);
         this.conflictingTable = conflictingTable;
+        this.args = List.copyOf(args);
     }
 
     public static TransactionSerializableConflictException create(
-            TableReference tableRef, long timestamp, long elapsedMillis) {
+            TableReference tableRef, long timestamp, long elapsedMillis, List<Arg<?>> args) {
         String msg = String.format(
                 "There was a read-write conflict on table %s.  This means that this table was marked as Serializable"
                         + " and another transaction wrote a different value than this transaction read.  startTs: %d "
                         + " elapsedMillis: %d",
                 tableRef.getQualifiedName(), timestamp, elapsedMillis);
-        return new TransactionSerializableConflictException(msg, tableRef);
+        return new TransactionSerializableConflictException(msg, tableRef, args);
+    }
+
+    @Override
+    public @Safe String getLogMessage() {
+        return "Transaction serializable conflict";
```

Comment:
Let's include more info here (something a bit more detailed, like the safe parts of the unsafe message)

Reply:
Log messages should generally be compile time constants. Any additional information should be provided in args.